packages/react-router/src/router.ts

```diff
@@ -163,6 +163,10 @@ export interface DehydratedRouterState {
   dehydratedMatches: DehydratedRouteMatch[]
 }
 
+export interface LoaderContext extends LoaderFnContext{
+  parentMatchPromise: Promise<void> | undefined
```

Comment:
I don't think this should be exposed as public API.
@tannerlinsley  which parts of `LoaderFnContext`should be exposed for codesplitting purpose?
see also #940